connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -175,9 +179,8 @@ protected <V> V execAndRetry(Operation<V> operation) throws Exception {
                 log.trace("Caught a retriable exception while executing {} operation with {}", context.stage(), context.executingClass());
                 errorHandlingMetrics.recordFailure();
                 if (checkRetry(startTime)) {
```

Comment:
I could be missing something, but do we even need `checkRetry`? Can we replace this with `time.milliseconds() < deadline`?

Asking because we've added special logic around `errorRetryTimeout` being negative in two different places, which seems unnecessary.

Reply:
Yeah, I think the lost test cases are fine as long as we still cover enough of the basic logic with more coarse-grained testing.

Reply:
Fair enough, I do think that the relevant logic is already tested by the `testExecAndHandle...` tests (I've added an additional one to check the max retries exceeded case)